### PR TITLE
Fix sort to use operators

### PR DESCRIPTION
**Pull request by vasukalariya**
# Description
Summarize the change.
- If an operator is specified sort has to be performed based on that.
- Consider operator for fetching the ranks.

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/segment/query/processor/sortcommand.go [hunk 1]
```diff
@@ -139,19 +139,33 @@ func compareString(a, b string) compare {
...
-		return NUMERIC
-	case segutils.SS_DT_STRING:
-		_, err := strconv.ParseFloat(CValEnc.CVal.(string), 64)
-		if err == nil {
-			// If floatValue is possible then it is considered as a number
+		switch op {
+		case "num", "auto", "":
+			return NUMERIC
+		case "str":
+			return STRING
```
Comment (nkunal) [inline pkg/segment/query/processor/sortcommand.go:151]: calling this enum as `STRING` is too generic, we should typically prefix some descriptor for such enums. for e.g. we call the `SS_DT_STRING` (sigscalr datatype string). 
Similarly we could call these things `RANK_STRING` etc.

File: pkg/segment/query/processor/sortcommand.go [hunk 2]
```diff
@@ -162,10 +176,10 @@ func getRank(CValEnc *segutils.CValueEnclosure) dTypeRank {
 }
 
 // Returns A comparison B
-func compareValues(valueA, valueB *segutils.CValueEnclosure, asc bool) compare {
+func compareValues(valueA, valueB *segutils.CValueEnclosure, asc bool, op string) compare {
 	var result compare
-	rankA := getRank(valueA)
-	rankB := getRank(valueB)
+	rankA := getRank(valueA, op)
```
Comment (nkunal) [inline pkg/segment/query/processor/sortcommand.go:181]: the `op` is specified during query time (or defaulted to some value). This `getRank` is getting called 2 x numRecords. The switch/case statement inside it also gets executed numRecords times.  When a explicit rank like str/num is passed in the query, then we should not waste time to find out the rank per record.

File: pkg/segment/query/processor/sortcommand.go (hunk 1 above)
Comment (vasukalariya) [reply pkg/segment/query/processor/sortcommand.go]: addressed in the latest commit.

**Review by AndrewHess — APPROVED** [review]
Nice tests

**Pull request merged**